diffsims/pattern/detector_functions.py

```diff
@@ -16,13 +16,48 @@
 # You should have received a copy of the GNU General Public License
 # along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
 
+import numpy as np
+
 from numpy.random import default_rng
 from scipy import ndimage as ndi
 
 
+def _process_seed_argument(seed):
+    """ Sets up a numpy random number generator with a seed"""
+    if seed is not None:
+        rng = default_rng(seed)
+    else:
+        rng = default_rng()
+
+    return rng
+
+
+def constrain_to_dynamic_range(pattern, detector_max=None):
```

Comment:
BTW, this can be handled by `skimage.exposure.rescale_intensity`, I believe: https://scikit-image.org/docs/stable/api/skimage.exposure.html#skimage.exposure.rescale_intensity.

Reply:
I did think skimage would have some of this code built in, but I've found that they can have a slightly strange set of habits about dtypes. Eg, that's why I've used numpy for my shot noise (which should be integer imo) code rather than the built skimage code. 

Also, as it happens the linked function does something slightly different to what we need. A value of 200 (detector_max set to 255) should be unchanged (for our purposes) whereas skimage would (if 200 was the max in the image) rescale it up to 255

Reply:
I agree that skimage doesn't have the solution all the time.

And I think I misunderstood your intent: you don't want to rescale anything, just clip values. Then you don't want that function. Sorry for the... noise.